rplidar.go

```diff
@@ -348,40 +382,50 @@ func (d *Device) scan(options lidar.ScanOptions) (lidar.Measurements, error) {
 
 			nodeAngle := (float64(node.GetAngle_z_q14()) * 90 / (1 << 14))
 			nodeDistance := float64(node.GetDist_mm_q2()) / 4
-			measurements = append(measurements, lidar.NewMeasurement(nodeAngle, nodeDistance/1000))
+
+			err := pc.Set(pointFrom(rdkUtils.DegToRad(nodeAngle), rdkUtils.DegToRad(0), float64(nodeDistance)/1000, 255))
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	if len(measurements) == 0 {
+	if pc.Size() == 0 {
 		return nil, nil
 	}
-	if options.NoFilter {
-		return measurements, nil
+
+	t_now := (time.Now())
+	t_now2 := t_now.Format(time.RFC3339)
+	t_str := strings.Replace(strings.Replace(t_now2, ":", "_", 3), "-", "", 2)
```

Comment:
Several things here. First, don't use snake_case in go, use camelCase. Second, no need for the parentheses around time.Now(). Third, go's time formatting is entirely customizable. Fourth, methods are composable here. Thus the exact same output can be had in one line:

`tStr := time.Now().Format("20060102T15_04_05Z")`

(Go uses a specific fixed date/time as the way to format things... this isn't actually hardcoding 2006)

Reply:
Interesting! Good to know James, I've always found string to time conversions a hassle and this seems a lot better

Reply:
Done.

Reply:
Actually - @Otterverse I am also trying to capture fractional seconds. I tried this:

`timeStamp.Format("20060102T15_04_05_00")`, but it doesn't seem to work for the fractional seconds. Can you see anything obviously wrong with this? And, in your example, what does the `Z` stand for?

Reply:
Z indicates "Zulu Time" (aka, UTC) in many timestamp formats, including RFC3339. I think I may have misunderstood golang's formatting slightly. I was adding the Z by itself, when really I needed to specify "Z0700" or "ZO7" or "Z07:00". It'll print the hours and minutes of the timezone offset for most timezones, but if the system is set to UTC, then it'll just print "Z"

Adding fractional seconds is just adding a decimal after the minutes and appending some zeros.

E.g. with four decimals: `Format("2006-01-02T15_04_05.0000Z0700")`
Right now on my PC (at UTC -0800) that'll print: `2022-02-01T18_31_12.2556-0800`

Go's time formatting is here: https://pkg.go.dev/time#pkg-constants and it's a little confusing as I mentioned. The problem is that you're trying to replace dashes with "" so that will break things potenailly, as there'd be no difference between UTC+8 and UTC-8 timezones. If you just want a usable filename that sorts in order, then force UTC and don't print the zone.

`Format("2006-01-02T15_04_05.0000")` >>> `2022-02-02T02_36_27.7745`